### PR TITLE
DYN-5152: Write a test to index into the result of a Replicated Function Call

### PR DESCRIPTION
**Pull request by sm6srw**
### Purpose

This pull request does:
* fix indexing from replicated function calls
* add a few micro tests

### Declarations

Check these if you believe they are true

- [ ] The codebase is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.
- [ ] This PR modifies some build requirements and the readme is updated

### Release Notes

(FILL ME IN) Brief description of the fix / enhancement. **Mandatory section** 

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: src/Engine/EmitMSIL/CodeGenIL.cs [hunk 1]
```diff
@@ -1159,7 +1159,7 @@ private void EmitArray<T>(Type arrType, IEnumerable<T> items, Action<T, int> ite
                     //can't handle these with compile time indexing.
                     //TODO remove IList from this if stmt when we figure out function call return wrapping behavior.
                     //this is still a problem for BuiltIn Dictionaries that are wrapped in an IList.
-                    if (arrayT == null || arrayT == typeof(IList) || arrayT == typeof(object))
+                    if (arrayT == null || arrayT == typeof(IList) || arrayT == typeof(object) || arrayT == typeof(DSASM.CLRStackValue))
```
Comment (pinzart90) [inline src/Engine/EmitMSIL/CodeGenIL.cs:1162]: so this means we will not support indexing into the result of a replicated call right ?

Comment (sm6srw) [reply src/Engine/EmitMSIL/CodeGenIL.cs:1162]: It means that we for now support it by calling the `valueAtIndex` method.

**Review by pinzart90 — APPROVED** [review]

**Comment by mjkkirschner** [comment]
@sm6srw this failed the tests, but I see the build machine seems to be failing for many jobs...

**Comment by sm6srw** [comment]
> @sm6srw this failed the tests, but I see the build machine seems to be failing for many jobs...

Yes, I started a rerun now (that I hope 003 will pick up).

**Pull request merged**